### PR TITLE
Implement Iterator#last to return last value of this iterator

### PR DESCRIPTION
**Pull request by lundibundi**
- [x] code is properly formatted (`npm run fmt`)
- [x] tests are added/updated
- [x] documentation is updated (`npm run doc` to regenerate documentation based on comments)
- [x] description of changes is added under the `Unreleased` header in CHANGELOG.md

### CODE REVIEW

**Comment by lundibundi** [comment]
ping @nechaido @belochub

File: lib/iterator.js [hunk 1]
```diff
@@ -233,6 +233,16 @@ class Iterator {
     return map;
   }
 
+  // Consumes an iterator returning last iterator value
```
Comment (SemenchenkoVitaliy) [inline lib/iterator.js:236]: Maybe
```suggestion
  // Consume an iterator and return last iterator value
```
since most comments use infinitive form?

Comment (lundibundi) [reply lib/iterator.js:236]: Current style is consistent with other such functions, we should change this for all of them if we want to.

**Comment by belochub** [comment]
Landed in https://github.com/metarhia/common/commit/827f2513a6ba112924bcfc1f4f84c9cd6fc6156c.

**Pull request closed without merging**